x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termenum/action/TermEnumResponse.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.termenum.action;
+
+import org.elasticsearch.action.support.DefaultShardOperationFailedException;
+import org.elasticsearch.action.support.broadcast.BroadcastResponse;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * The response of the termenum/list action.
+ */
+public class TermEnumResponse extends BroadcastResponse {
+
+    public static final String TERMS_FIELD = "terms";
+    public static final String TIMED_OUT_FIELD = "timed_out";
+    public static final String SKIPPED_FIELD = "skipped";
```

Comment:
what does `SKIPPED_FIELD` is responsible for ?

Reply:
Like the search API - it's what shards failed the canMatch phase.
If you supply an `index_filter` it does a canMatch on each shard before considering the terms in it. (Currently the test fails for use of index filters after I moved the code to xpack. Something about QueryBuilders not being available).

Reply:
Removed